### DISCUSSION TITLE
convert anim gif to pdf

### DISCUSSION THREAD

**Discussion by michaelaye** [quarto-dev/quarto-cli#3551] [Q&A]
Hello!
First, thanks for the nice tool!
I have a revealjs qmd file that I want to render as pdf.
It contains an animated .gif file.
When I render it with pandoc directly:

```bash
❯ pandoc -r markdown index.qmd -o pres.pdf

talks/2022-fub-seminar/presentation on  main [!?] via 🅒 py310 took 5s
```

it does something without complaining, and it simply takes the first frame of the animated gif and places it into the pdf. That's what I want to happen. 
However, the layout and figure sizes are all wrong and clipped, so I'm trying to render it with quarto instead, hoping that the resulting pdf would look better.

When quarto->xelatex hits the animated gif I get:

```bash
❯ quarto render index.qmd --to pdf
[1/2] talks/2022-fub-seminar/presentation/index.qmd
running xelatex - 1
  This is XeTeX, Version 3.141592653-2.6-0.999994 (TeX Live 2022) (preloaded format=xelatex)
   restricted \write18 enabled.
  entering extended mode

updating tlmgr

updating existing packages

compilation failed- error
LaTeX Error: Cannot determine size of graphic in image13.gif (no BoundingBox)
.

See the LaTeX manual or LaTeX Companion for explanation.
Type  H <return>  for immediate help.
 ...

l.470 {\centering \includegraphics{image13.gif}


see /Users/klay6683/Dropbox/src/quarto_site/talks/2022-fub-seminar/presentation/index.log for more information.
```

Does someone know how I can get this to work?

Also, is there a way to configure quarto to always place a PDF version of a presentation next to a link to the executable revealjs presentation on a website?

**Comment by mroberts1** [comment]
I am having the exact same problem - tried using pdflatex as pdf-engine but the result is the same. Can LaTex not process gifs?

**Comment by mcanouil** [comment]
Reveal.js presentation are not converted to PDF via Pandoc or Quarto as you expect.
PDF uses LaTeX while Reveal.js is a HTML framework.
What Pandoc does is changing the format from revealjs to pdf which of course ends up with something different.


If you want an actual PDF version of your Reveal.js HTML presentation, see <https://quarto.org/docs/presentations/revealjs/presenting.html#print-to-pdf>.

For R user, see code below.
Note that similar approach is possible for other language.

```r
quarto::quarto_render("myslides.qmd")
pagedown::chrome_print("myslides.html")
```

This (or similar) can be added as a post-render script, see <https://quarto.org/docs/projects/scripts.html#pre-and-post-render>.

**Comment by tomkom** [comment]
I am encountering the same problem, but not when printing slides, but wanting to export a pdf version of a book at the same time as html content. Would this behavior be possible? I currently need to provide alternative content for all animated gifs....

**Reply by cderv** [reply]
To include such content in a PDF document, you need to use specific tools. The [`media9` CTAN package](https://www.ctan.org/pkg/media9) could be useful or more classically the [animate package](https://ctan.org/pkg/animate). 

I don't think Pandoc does not support those type by default (like https://github.com/jgm/pandoc/issues/7181) and so the LaTeX writer won't handle GIF specifically. This need to be done at the user level. 

This could be done as a Quarto extension using Lua filter to write the correct LaTeX in the resulting document. Without this, [conditional content](https://quarto.org/docs/authoring/conditional.html) are a way to write content for HTML and LaTeX differently. 

Don't know well the LaTeX command to do that but seems possible

**Reply by mbcann01** [reply]
First of all, thank you dev team for this amazing tool! Second, I'm having a similar issue. I'm creating a Quarto book that includes some gifs. Some of my readers have also asked for a PDF version of the book. I was hoping that the PDF version would just display the first frame of the PDF by default, but no such luck. Add that behavior natively, or the ability to choose that behavior in _quarto.yml is definitely a wish list item for me. In the meantime, I'll look into some of the workarounds you've suggested. Thanks again!!
